Modules/Module_Processing/Module_ASL/xASL_wrp_Quantify.m

```diff
@@ -482,7 +434,7 @@ function xASL_wrp_Quantify(x, PWI4D_Path, pathOutputCBF, M0Path, SliceGradientPa
 
 %% ------------------------------------------------------------------------------------------------
 %% 11.  Create standard space masked image to visualize masking effect
-if xASL_exist(x.P.Pop_Path_qCBF, 'file') && (strcmp(pathOutputCBF, x.P.Pop_Path_qCBF) || x.Q.bUseBasilQuantification)
+if xASL_exist(x.P.Pop_Path_qCBF, 'file') && (strcmp(pathOutputCBF, x.P.Pop_Path_qCBF) || x.modules.asl.bUseBasilQuantification) && ~bSaveCBF4D
```

Comment:
bSaveCBF4D -> I understand that BASIL doesn't output this. But then we should issue a warning about this, and say that for saving CBF4D you need the non-BASIL quantification

masking: here we created a masked version of the standard space qCBF, which we can still (and should still) do for BASIL/FABBER output

Reply:
1. We already have such a warning in xASL_module.ASL line 434:
    warning('Saving CBF4D was requested but not implemented yet for multi-PLD or multi-TE, skipping');
    So we don't need a second warning here, just a safety check.
    
 2. OK, mask for BASIL output -> please make a more specific comment of what to do and where.

Reply:
1:
My point is: we have several checks for setting combinations. If there is an invalid setting combination, we issue a warning, and change the setting. We do this ideally for all settings at a single location, but at least we don't do it twice. So choose if you want this here or in xASL_module_ASL, and change throughout.
The reason is that it is nicely modular/readable. If we do a setting check at a location where we only throw a warning but not change the setting, and not give an explanation, this is confusing and makes our code less readable. Let's have this all at a single location.

So we have the A setting check B warning C setting change at a single location, and remove this elsewhere.

I would probably be in favor of having this for all combinations at the quantification start itself, because it is closest to the action. I know that for quantification parameters like PLD, TE, LD we have this at the start of xASL_module_ASL, but that is because we need those settings throughout the whole pipeline. But I would also be fine to move these settings to the start of xASL_module_ASL, but at least at a single location.


2: Here we create a qCBF_masked. If BASIL is run, we still need this (but then create this when qCBF is already created, this should be similar to without BASIL). When we have bSaveCBF4D, we ALSO save qCBF so also qCBF_masked. So saving qCBF4D doesn't exclude saving qCBF and qCBF_masked.

-> qCBF_masked is used for population module-> CBF ROI values, as it removes vascular, whereas qCBF is used for spatial CoV ROI values. It is nice to keep these NIfTIs so users can see what was masked out for the ROI values (but not necessary for archiving)

Reply:
1. OK. I get your point now. And agree that a single location for warnings is a must. Actually, the best location is probably the start of xASL_module_ASL in the function where we check all parameters. I'll make the check there, input the warning and then switch this setting of. In the original place of the code, I'll leave only safety catch without an explanation for the user - just in case somebody messes up. Done in 44aa553b

2. Irrespective of bSaveCBF4D, we always save qCBF and qCBF_masked - this happens in the step above. Without BASIL we do it for both native and standard. With BASIL, we do it for native only, but then transform this to standard. At least it should be like this unless there is some bug. So things should actually already work as you want. If you feel that some intermediate image is delete, then I agree that we don't have to delete it. Not sure if this answers your comments, but I don't really understand what the issue is here.